gobrew.go

```diff
@@ -84,6 +84,104 @@ func NewGoBrewDirectory(homeDir string) GoBrew {
 	return gb
 }
 
+func (gb *GoBrew) Interactive(ask bool) {
+	currentVersion := gb.CurrentVersion()
+	currentMajorVersion := ExtractMajorVersion(currentVersion)
+
+	latestVersion := gb.getLatestVersion()
+	latestMajorVersion := ExtractMajorVersion(latestVersion)
+
+	modVersion := gb.getModVersion()
+
+	if modVersion == "" {
+		modVersion = "None"
+	}
+
+	fmt.Println()
+
+	if currentVersion == "" {
+		currentVersion = "None"
+		color.Warnln("GO Installed Version", ".......", currentVersion)
+	} else {
+		labels := []string{}
+		if modVersion != "None" && currentMajorVersion != modVersion {
+			labels = append(labels, "not same as go.mod")
+		}
+		if currentVersion != latestVersion {
+			labels = append(labels, "not latest")
+		}
+		label := ""
+		if len(labels) > 0 {
+			label = "(" + strings.Join(labels, ", ") + ")"
+		}
+		if label != "" {
+			label = " " + color.FgRed.Render(label)
+		}
+		color.Successln("GO Installed Version", ".......", currentVersion+label)
+	}
+
+	if latestMajorVersion != modVersion {
+		label := " " + color.FgYellow.Render("(not latest)")
+		color.Successln("GO go.mod Version", "   .......", modVersion+label)
+	} else {
+		color.Successln("GO go.mod Version", "   .......", modVersion)
+	}
+
+	color.Successln("GO Latest Version", "   .......", latestVersion)
+	fmt.Println()
+
+	if currentVersion == "None" {
+		color.Warnln("GO is not installed.")
+		c := true
+		if ask {
+			c = AskForConfirmation("Do you want to use latest GO version (" + latestVersion + ")?")
+		}
+		if c {
+			gb.Install(latestVersion)
+			gb.Use(latestVersion)
+		}
+		return
+	}
+
+	if currentMajorVersion != modVersion {
+		color.Warnf("GO Installed Version (%s) and go.mod Version (%s) are different.\n", currentMajorVersion, modVersion)
+		c := true
+		if ask {
+			c = AskForConfirmation("Do you want to use GO version same as go.mod version (" + modVersion + "@latest)?")
+		}
+		if c {
+			gb.Install(modVersion + "@latest")
```

Comment:
`modVersion` and `@latest` is different versions. Is`n it?

Reply:
`modVersion` is usually the major version like `1.20` or `1.21`.
In order to install the mod version, latest of the major version, `1.20@latest` or `1.21@latest` is required.
Since, just passing the `modVersion` will end up installing the `1.21.0`

Reply:
Understood thanks!

Reply:
Thank you very much for your review and check too. Appreciate it.